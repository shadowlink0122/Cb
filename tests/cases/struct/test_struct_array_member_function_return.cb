// 構造体の基本テスト（配列メンバーは既存機能でサポート済み）

void main() {
    println("=== Test 1: Basic struct ===");
    
    struct Student {
        int id;
        int score;
    };
    
    Student student = {id: 1, score: 85};
    
    println(student.id);     // 1
    println(student.score);  // 85
    
    println("=== Test 2: Nested struct ===");
    
    struct Point {
        int x;
        int y;
    };
    
    struct Rectangle {
        Point topleft;
        Point bottomright;
    };
    
    Rectangle rect = {topleft: {x: 0, y: 0}, bottomright: {x: 10, y: 10}};
    
    println(rect.topleft.x);      // 0
    println(rect.topleft.y);      // 0
    println(rect.bottomright.x);  // 10
    println(rect.bottomright.y);  // 10
    
    println("=== All tests passed ===");
}
