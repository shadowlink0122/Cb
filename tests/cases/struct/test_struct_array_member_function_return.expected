=== Test 1: Basic struct ===
1
85
=== Test 2: Nested struct ===
0
0
10
10
=== All tests passed ===
